crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -328,6 +328,9 @@ pub unsafe trait WorldQuery {
     /// constructing [`Self::Fetch`](crate::query::WorldQuery::Fetch).
     type State: Send + Sync + Sized;
 
+    /// Runtime config value passed to [`WorldQuery::init_state`]
+    type Config;
```

Comment:
Is there any use case for this to *not* have a `Default` trait?

It seems like we could significantly clean up this PR by just swapping to a trait bound here.

Reply:
If you wanted to implement a dynamic query, you would want to pass as `Config = ComponentId` and `ComponentId` would not implement `Default`.